lib/ReportFunctions.groovy

```diff
@@ -0,0 +1,160 @@
+
+
+
+
+class ReportFunctions{
+   
+/*
+    enum FuncType{
+        GENERIC("GENERIC"),
+        CONTIG("CONTIG"),
+        READQUALITY("READQUALITY"),
+        AUTOFAIL("AUTOFAIL"),
+        COVERAGE("COVERAGE")
+         
+    }
+*/
+
+
+    enum FuncType{
+        GENERIC,
+        CONTIG,
+        READQUALITY,
+        AUTOFAIL,
+        COVERAGE 
+    }
+
+
+    static def select_qc_func(java.util.LinkedHashMap qual_data, java.lang.String metric, java.util.ArrayList qc_message, java.util.LinkedHashMap meta_info, java.lang.String func){
+        def check_failed = 0
+        def reisolate = 0
+        def resequence = 0
+        def check_ignored = 0
+        def failed_p = false
+        def checks = 0
+        def function = func as FuncType
+
+        switch(function){
+            case FuncType.GENERIC:
+                (checks,
+                reisolate, 
+                resequence, 
+                failed_p, 
+                check_failed, 
+                check_ignored) = this.generic_qc_func(qual_data, metric, qc_message)
```

Comment:
I would use `ReportFunctions.generic_qc_func()` to call the function (or just `generic_qc_func()`) since these are static functions and `this` is meant to refer to the particular instance of a class (for non-static functions). Though it seems like it works anyways like how you have things.